value/matrix.go

```diff
@@ -410,47 +410,81 @@ func (m *Matrix) vrotate(n int) Value {
 }
 
 // transpose returns (as a new matrix) the transposition of the argument.
-func (m *Matrix) transpose() *Matrix {
-	rank := m.Rank()
-	if rank < 2 {
-		// Shouldn't happen but easy.
-		return m.Copy()
-	}
-	// Shapes as integers not values are easier here.
-	// The reversed shape of m, that is, shape of m transposed.
+func (m *Matrix) transpose(c Context) *Matrix {
+	v := make(Vector, m.Rank())
+	origin := c.Config().Origin()
+	for i := range v {
+		v[len(v)-1-i] = Int(i + origin)
+	}
+	return m.dtranspose(c, v)
+}
+
+// dtranspose returns the dyadic transpose of m specified by v,
```

Comment:
for reasons lost to history, ivy uses unary and binary rather than monastic (ha) and dyadic. please adjust.

Reply:
Done. I did leave one note here saying 'APL calls this operator the dyadic transpose', because that's important for finding documentation about it with a web search.